packages/mgt-components/src/components/mgt-todo/mgt-todo.ts

```diff
@@ -152,123 +169,141 @@ export class MgtTodo extends MgtTasksBase {
-      <span class="NewTaskDue">
-        ${this.renderCalendarIcon()}
-        <input
-          type="date"
-          label="new-taskDate-input"
-          aria-label="new-taskDate-input"
-          role="textbox"
-          .value="${this.dateToInputValue(this._newTaskDueDate)}"
-          @change="${(e: Event) => {
-            const value = (e.target as HTMLInputElement).value;
-            if (value) {
-              this._newTaskDueDate = new Date(value + 'T17:00');
-            } else {
-              this._newTaskDueDate = null;
-            }
-          }}"
-        />
-      </span>
-    `;
+    this._isNewTaskBeingAdded = true;
+    this.requestUpdate();
 
-    return html`
-      ${taskList} ${taskDue}
-    `;
+    try {
+      await this.createNewTask();
+    } finally {
+      this.clearNewTaskData();
+      this._isNewTaskBeingAdded = false;
+      this.requestUpdate();
+    }
   }
 
   /**
-   * Render the header part of the component.
+   * Render the panel for creating a new task
    *
    * @protected
-   * @returns
+   * @returns {TemplateResult}
    * @memberof MgtTodo
    */
-  protected renderHeaderContent(): TemplateResult {
-    if (this.isLoadingState) {
-      return html`
-        <div class="header__loading"></div>
-      `;
-    }
+  protected renderNewTask(): TemplateResult {
+    const addIcon = html`
+      <fluent-button 
+        class="task-add-icon" 
+        @click="${() => this.addTask()}">
+        ${getSvg(SvgIcon.Add)}
+      </fluent-button>
+    `;
 
-    const lists = this._lists || [];
-    const currentList = this._currentList;
-    const targetId = this.targetId;
-    let listSelect: TemplateResult;
-
-    if (targetId && lists.length) {
-      const list = lists.find(l => l.id === targetId);
-      if (list) {
-        listSelect = html`
-          <span class="PlanTitle">
-            ${list.displayName}
-          </span>
-        `;
-      }
-    } else if (currentList) {
-      const listOptions = {};
-      for (const l of lists) {
-        listOptions[l.displayName] = () => this.loadTaskList(l);
-      }
+    const cancelIcon = html`
+      <fluent-button
+        class="task-cancel-icon" 
+        @click="${() => this.clearNewTaskData()}">
+        ${getSvg(SvgIcon.Cancel)}
+      </fluent-button>
+    `;
 
-      listSelect = mgtHtml`
-        <mgt-arrow-options .value="${currentList.displayName}" .options="${listOptions}"></mgt-arrow-options>
-      `;
-    }
+    const calendarTemplate = html`
+      <fluent-text-field
+        type="date"
+        id="new-taskDate-input"
+        class="date"
+        aria-label="${this.strings.newTaskDateInputLabel}"
+        .value="${this.dateToInputValue(this._newTaskDueDate)}"
+        @change="${(e: Event) => {
+          const value = (e.target as HTMLInputElement).value;
+          if (value) {
+            this._newTaskDueDate = new Date(value + 'T17:00');
+          } else {
+            this._newTaskDueDate = null;
+          }
+        }}">
```

Comment:
Move this to a class method then you reference it like `@change=${this.handleChange}`. Avoids creation of an anonymous function on render.